tests/gocase/tls/tls_test.go

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package tls
+
+import (
+	"context"
+	"crypto/tls"
+	"testing"
+
+	"github.com/apache/incubator-kvrocks/tests/gocase/util"
+	"github.com/go-redis/redis/v9"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTLS(t *testing.T) {
+	if !util.TLSEnable() {
+		t.Skip("TLS tests run only if tls enabled.")
+	}
+
+	ctx := context.Background()
+
+	srv := util.StartTLSServer(t, map[string]string{})
+	defer srv.Close()
+
+	defaultTLSConfig, err := util.DefaultTLSConfig()
+	require.NoError(t, err)
+
+	rdb := srv.NewClientWithOption(&redis.Options{TLSConfig: defaultTLSConfig, Addr: srv.TLSAddr()})
+	defer func() { require.NoError(t, rdb.Close()) }()
+
+	doWithTLSClient := func(tlsConfig *tls.Config, f func(c *redis.Client)) {
+		c := srv.NewClientWithOption(&redis.Options{TLSConfig: tlsConfig, Addr: srv.TLSAddr()})
+		defer func() { require.NoError(t, c.Close()) }()
+		f(c)
+	}
+
+	t.Run("TLS: Not accepting non-TLS connections on a TLS port", func(t *testing.T) {
+		c := srv.NewClientWithOption(&redis.Options{TLSConfig: defaultTLSConfig})
```

Comment:
```suggestion
		c := srv.NewClientWithOption(&redis.Options{Addr: srv.TLSAddr()})
```
I think the default port is a non-TLS port, and it is a non-TLS connection so we do not need TLSConfig here.

Reply:
Good catch.